File: lib/business/models/gestion/application.dart
import 'package:immobilx/business/models/user/user.dart';

class ApplicationModel {
  final String id;
  final int propertyId;
  final int tenantId;
  final String? message;
  final String status;
  final DateTime createdAt;
  final User? tenant;

  ApplicationModel({
    required this.id,
    required this.propertyId,
    required this.tenantId,
    required this.message,
    required this.status,
    required this.createdAt,
    this.tenant,
  });

  factory ApplicationModel.fromJson(Map<String, dynamic> json) {
    return ApplicationModel(
      id: json['id'].toString(),
      propertyId: int.parse(json['propertyId'].toString()),
      tenantId: int.parse(json['tenantId'].toString()),
      message: json['message'],
      status: json['status'],
      createdAt: DateTime.parse(json['createdAt']),
      tenant: json['tenant'] != null ? User.fromJson(json['tenant']) : null,
    );
  }
}

File: back/app/models/application.ts
import { DateTime } from 'luxon'
import { BaseModel, belongsTo, column } from '@adonisjs/lucid/orm'
import type { BelongsTo } from '@adonisjs/lucid/types/relations'
import Property from '#models/property'
import User from '#models/user'

export enum ApplicationStatus {
  PENDING = 'pending',
  ACCEPTED = 'accepted',
  REJECTED = 'rejected',
}

export default class Application extends BaseModel {
  @column({ isPrimary: true })
  declare id: number

  @column()
  declare propertyId: number

  @column()
  declare tenantId: number

  @column()
  declare message: string | null

  @column()
  declare status: ApplicationStatus

  @column.dateTime({ autoCreate: true })
  declare createdAt: DateTime

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  declare updatedAt: DateTime

  @belongsTo(() => Property)
  declare property: BelongsTo<typeof Property>

  @belongsTo(() => User, { foreignKey: 'tenantId' })
  declare tenant: BelongsTo<typeof User>
}




